### ISSUE TITLE
New Update on 20/10/21 seems to have broken the patch

### ISSUE THREAD

**Issue by nes4day** [mpm11011/ac7-ultrawide#22]
There was a patch earlier today that allegedly removed Denuvo, and after running the script, Steam reports a game load error.
![PY](https://user-images.githubusercontent.com/44482642/138013616-ae1fcc1b-0bd4-4fd3-9924-331fcb6b6c5b.PNG)

**Comment by WinkelCode** [comment]
Seems like the update broke stuff. I'll look into it.

**Comment by WinkelCode** [comment]
Good news! Seems like despite removing Denuvo, the new DRM is more annoying... at least for us :)

The solution is to simply remove the new Steam DRM from the exe: https://github.com/atom0s/Steamless (Compile it yourself or grab the latest build from the "Releases" button on the right side)

FYI: You can copy the extracted "Steamless" folder and the Ace7Game.exe both to a Windows Sandbox window, you can remove the DRM there, in case you want to do that. (I don't have any reason to mistrust Steamless, but since it works.)

Make sure to rename the extracted .exe back to "Ace7Game.exe" and put it back in the original directory. IMPORTANT: Make sure it's not `Ace7Game.exe.exe`, enable file extensions in Explorer if you haven't already (you REALLY should!)

I will be looking into automating this or improving the workflow in other ways, but for now this should work.

I tested the into sequence and everything looked OK, though I *think* the FOV might be higher than it should be (I'll look into that later). Since we're removing DRM (and FOV might be too high in the current version), I'd hold off on playing multiplayer for now. I don't really care about AC7 multiplayer myself so I might just try it later and see if I get banned by the anticheat.

**Comment by nes4day** [comment]
That worked really well, did some quick testing and found no issues, thank you. Another question if you don't mind, was the old-issue with HUD alignment in Minigame Cutscenes (Refuel, Landing/Take-off; see attached) ever resolved? 

![11](https://user-images.githubusercontent.com/44482642/138087978-6650a910-9957-4393-9e55-bccbd613e46a.png)

**Comment by WinkelCode** [comment]
I am not aware of any fixes regarding this issue, the main part of the 3Dmigoto fix is generic and automatically finds elements to readjust. You can read more here (if you speak regex): https://github.com/mpm11011/ac7-ultrawide/blob/master/Mods/hudtextfix.ini

Since the icon to skip and the subtitles are both shifted, and I think they are separate elements, the fix may be thinking it's not "$in_game".

It's been a while but I believe I was the one who wrote the "$in_game" part and it was mostly just randomly finding a shader that's there in the menus but not the game (menus center fine usually).

I've been meaning to look into remaking some parts of the fix to address common problems people are facing so this is something to add to the list, maybe there is a better shader or method to do this with.

I believe if you open the pause menu you can actually see some parts of the HUD shifting during the fade, that's related to the whole "$in_game" thing.

Edit: Just noticed, "$in_game" had "TODO - better way to do this?" in a comment since October 2019, classic.

**Comment by nes4day** [comment]
I played around with that INI in the past, but it came to nothing. Thank you again for helping me on this, hopefully you will find some better way to implement fixes, but either way, the game is perfectly playable with the current fix.

**Comment by WinkelCode** [comment]
If the option is available for you, please reopen the issue since the core problem isn't fixed.

**Comment by nes4day** [comment]
Done, reopened

**Comment by erockg74** [comment]
New to the AC series. Just bought AC7 on 2021-12-28 excited to play on new Samsung Odyssey Neo G9. Shocked AC7 isn't 32:9 out of the box. Installed and executed Seamless on AC7 exe file. Installed Python and downloaded 3Dmigoto-1.3.16.zip. Followed the instructions and troubleshooting from various sites. Works fantastically! AC7 is much more fun in glorious 5440x1440. Thanks!

**Comment by QuadReaktor** [comment]
Today or yesterday was a new update I dont know if it don't works because of that.
I have this error: here a picture. I hope anyone have a fix that we can play AC7 again in Utrawidescreen :
![Error](https://user-images.githubusercontent.com/89872909/150419540-4429bc2c-fae0-44e5-a2d3-709933eb77bd.jpg)

**Comment by WinkelCode** [comment]
I believe that error comes from the DRM when the executable is modified, did you remove the DRM with Steamless first?

Edit: You'll also have to use Steamless after every update that changes the executable (usually every update).

**Comment by QuadReaktor** [comment]
Yeah now it works, thanks a lot !!

Am Fr., 21. Jan. 2022 um 00:11 Uhr schrieb martinsstuff <
***@***.***>:

**Comment by Marshara** [comment]
Hello everyone, I'm completely new to this, which is why I would like to ask for a bit of guidance. I have followed your guide and ended with the same error as everyone else, but here I'm lost. I downloaded Steamless and unpacked it, grabbed Ace7Game.exe file and threw it into the unpacked folder and expected magic... I was obviously wrong haha. Could please someone guide me how to apply Steamless to AC7.exe file properly?

Update: managed to find a version of Steamless with .exe file, works now. The only question left is: I have UI on my left monitor, which isn't exactly a good thing... :D did anyone have a similar issue and solved it? 

Thank you in advance and have a nice day :)

**Comment by WinkelCode** [comment]
@Marshara 

1. Use Steamless to remove the DRM from ace7game.exe
2. Remove/rename the original ace7game.exe and replace it with the DRM-free one.
3. Download the repository: https://github.com/mpm11011/ac7-ultrawide/archive/refs/heads/master.zip
4. Unpack it and place all the files into the ACE COMBAT 7 directory so that magic.py is in the same folder as ace7game.exe
5. Download 3dmigoto: https://github.com/bo3b/3Dmigoto/releases/download/1.3.16/3Dmigoto-1.3.16.zip
6. Move the .zip into the same folder as magic.py and ace7game.exe (don't unpack it)
7. Run magic.py

Before you try this you might want to remove everything in the ACE COMBAT 7 directory except the folders "Engine" and "Game". Then verify the game files using Steam (https://help.steampowered.com/en/faqs/view/0C48-FCBD-DA71-93EB). This will ensure your files are unmodified before you start.

**Comment by QuadReaktor** [comment]
*I have done all this :*

*Use Steamless to remove the DRM from ace7game.exe*
> *Remove/rename the original ace7game.exe and replace it with the DRM-free
> one.*
> *Download the repository:
> https://github.com/mpm11011/ac7-ultrawide/archive/refs/heads/master.zip
> <https://github.com/mpm11011/ac7-ultrawide/archive/refs/heads/master.zip>*
> *Unpack it and place all the files into the ACE COMBAT 7 directory so that
> magic.py is in the same folder as ace7game.exe*
> *Download 3dmigoto:
> https://github.com/bo3b/3Dmigoto/releases/download/1.3.16/3Dmigoto-1.3.16.zip
> <https://github.com/bo3b/3Dmigoto/releases/download/1.3.16/3Dmigoto-1.3.16.zip>*
> *Move the .zip into the same folder as magic.py and ace7game.exe (don't
> unpack it)*
> *Run magic.py*
>
> *Before you try this you might want to remove everything in the ACE COMBAT
> 7 directory except the folders "Engine" and "Game". Then verify the game
> files using Steam
> (https://help.steampowered.com/en/faqs/view/0C48-FCBD-DA71-93EB
> <https://help.steampowered.com/en/faqs/view/0C48-FCBD-DA71-93EB>). This
> will ensure your files are unmodified before you start.*
>
After I done all this like I used to do everytime, but this time I get this
error:



Am Di., 1. März 2022 um 09:22 Uhr schrieb Marshara ***@***.***

**Comment by QuadReaktor** [comment]
I forgot to upload the Picture ! Magig.py works fine.
 [image: magic.jpg]
but after start the Game I get this error
[image: error.jpg]
😰 I hope there will be a solution 1






Am Do., 3. März 2022 um 12:10 Uhr schrieb Quad ***@***.***

**Comment by Marshara** [comment]
@martinsstuff 

Thank you. I followed as you instructed. Deleted everything but Engine and Game folders, and followed the exact order and it works now.

**Comment by QuadReaktor** [comment]
Yeah I now it works for me too I forgot to rename the exe + seamless.
Thanks a lot for helping !!

Am Do., 3. März 2022 um 12:44 Uhr schrieb Marshara ***@***.***

**Comment by BRuh420CMd** [comment]
can someone tell me how to use steamless please cause im lost

edit: bruh i just realized i was installing the fucking source code, i figured it out now

**Comment by DAOWAce** [comment]
If anyone's still following this, the patcher is currently busted again due to an update in May 2022.

Gonna need help getting it working again.
